cloudant-client/src/test/java/com/cloudant/tests/HttpTest.java

```diff
@@ -996,4 +996,44 @@ public MockResponse dispatch(RecordedRequest request) throws InterruptedExceptio
         assertEquals("The third request should have been for /", "/",
                 MockWebServerResources.takeRequestWithTimeout(mockWebServer).getPath());
     }
+
+    /**
+     * Test that having no body and hence no error stream correctly results in the 403 causing
+     * a CouchDbException with no NPEs.
+     *
+     * @throws Exception
+     */
+    @Test(expected=CouchDbException.class)
+    public void noErrorStream403() throws Exception {
+        mockWebServer.enqueue(MockWebServerResources.OK_COOKIE);
+        mockWebServer.enqueue(new MockResponse().setResponseCode(403));
+        CloudantClient c = CloudantClientHelper.newMockWebServerClientBuilder(mockWebServer)
+                .username("a")
+                .password("b")
+                .build();
+
+        String response = c.executeRequest(Http.GET(c.getBaseUri())).responseAsString();
+        fail("There should be an exception, but received response " + response);
+    }
+
+    /**
+     * Test that having no body and hence no error stream correctly results in a 401 cookie renewal.
```

Comment:
Actually I think I understand what is going on here but it's not always when the enqueued responses will be streamed, especially because they are all declared at the top. Perhaps some comments around each group of enqueues might help maintainability.